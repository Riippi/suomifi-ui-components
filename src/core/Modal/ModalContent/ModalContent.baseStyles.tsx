import { css } from 'styled-components';
import { defaultThemeTokens as theme } from '../../theme';

export const baseStyles = css`
  &.fi-modal_content {
    position: relative;
    flex: 1 1 auto;
    & .fi-modal_content_wrapper {
      overflow-y: auto;
      border-bottom: 1px solid ${theme.colors.depthLight1};
      padding: 24px ${theme.spacing.xl} 50px ${theme.spacing.xl};
    }
    & .fi-modal_content_gradient-overlay {
      height: 50px;
      width: calc(100% - 40px);
      position: absolute;
      top: calc(100% - 51px);
      margin: 0 ${theme.spacing.m};
      background: linear-gradient(
        0deg,
        ${theme.colors.whiteBase},
        rgba(255, 255, 255, 0)
      );
    }

    &--no-scroll {
      & .fi-modal_content_wrapper {
        overflow: hidden;
        border-bottom: none;
        padding: 34px ${theme.spacing.xl} 10px ${theme.spacing.xl};
      }
    }

    &--small-screen {
      & .fi-modal_content_wrapper {
        padding-top: ${theme.spacing.m};
        padding-left: ${theme.spacing.m};
        padding-bottom: 50px;
        padding-right: ${theme.spacing.m};
      }
    }
  }
`;
