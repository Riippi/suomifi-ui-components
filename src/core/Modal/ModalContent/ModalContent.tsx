import React, { Component, ReactNode } from 'react';
import { default as styled } from 'styled-components';
import classnames from 'classnames';
import { HtmlDiv, HtmlDivProps } from '../../../reset';
import { ModalConsumer, ModalVariant, baseClassName } from '../Modal/Modal';
import { baseStyles } from './ModalContent.baseStyles';

export interface ModalContentProps
  extends Omit<HtmlDivProps, 'children' | 'className'> {
  /** Modal container div class name for custom styling. */
  className?: string;
  /** Children */
  children: ReactNode;
}

interface InternalModalContentProps extends ModalContentProps {
  id?: string;
  modalVariant: ModalVariant;
  scrollable: boolean;
}

const contentClassName = `${baseClassName}_content`;
const modalContentClassNames = {
  smallScreen: `${contentClassName}--small-screen`,
  noScroll: `${contentClassName}--no-scroll`,
  heading: `${contentClassName}_heading`,
};

class BaseModalContent extends Component<InternalModalContentProps> {
  render() {
    const {
      className,
      title,
      children,
      scrollable,
      modalVariant,
      ...passProps
    } = this.props;

    return (
      <HtmlDiv
        className={classnames(contentClassName, className, {
          [modalContentClassNames.noScroll]: scrollable === false,
          [modalContentClassNames.smallScreen]: modalVariant === 'smallScreen',
        })}
        {...passProps}
      >
        {children}
      </HtmlDiv>
    );
  }
}

const StyledModalContent = styled(BaseModalContent)`
  ${baseStyles}
`;

/**
 * <i class="semantics" />
 * Use for showing modal content.
 * Props other than specified explicitly are passed on to outermost content div.
 */
export class ModalContent extends Component<ModalContentProps> {
  render() {
    return (
      <ModalConsumer>
        {({ variant, scrollable }) => (
          <StyledModalContent
            modalVariant={variant}
            scrollable={scrollable}
            {...this.props}
          />
        )}
      </ModalConsumer>
    );
  }
}
