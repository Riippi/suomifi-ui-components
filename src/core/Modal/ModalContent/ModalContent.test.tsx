import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { render } from '@testing-library/react';
import { ModalContent, ModalContentProps } from './ModalContent';
import { ModalTitle, Modal } from '../';

let appRoot: HTMLDivElement | null = null;

beforeEach(() => {
  appRoot = document.createElement('div');
  appRoot.setAttribute('id', 'root');
  document.body.appendChild(appRoot);
});

afterEach(() => {
  if (!!appRoot) {
    unmountComponentAtNode(appRoot);
    appRoot.remove();
    appRoot = null;
  }
});

describe('Basic ModalContent', () => {
  const text = 'Modal Content';

  const BasicModal = (props?: Partial<ModalContentProps>) => (
    <Modal appElementId="root" visible={true}>
      <ModalContent {...props} data-testid="modal-content-id">
        <ModalTitle>Test modal</ModalTitle>
        <p>{text}</p>
        <button>x</button>
      </ModalContent>
    </Modal>
  );

  it('should have given title', () => {
    const { getByText } = render(BasicModal());
    expect(getByText('Test modal')).toBeTruthy();
  });

  it('should have given children', () => {
    const { getByText } = render(BasicModal());
    expect(getByText(text)).toBeTruthy();
  });

  it('should have given className', () => {
    const { getByTestId } = render(
      BasicModal({ className: 'test-class-name' }),
    );
    expect(getByTestId('modal-content-id')).toHaveClass('test-class-name');
  });

  it('should match snapshot', () => {
    const { baseElement } = render(BasicModal());
    expect(baseElement).toMatchSnapshot();
  });
});

describe('ModalContent variant', () => {
  const SmallScreenModal = (
    <Modal appElementId="root" visible={true} variant="smallScreen">
      <ModalContent data-testid="modal-content-id">
        <ModalTitle>Test modal</ModalTitle>
        <p>Modal Content</p>
        <button>x</button>
      </ModalContent>
    </Modal>
  );

  it('smallScreen should have correct classname', () => {
    const { getByTestId } = render(SmallScreenModal);
    expect(getByTestId('modal-content-id')).toHaveClass(
      'fi-modal_content--small-screen',
    );
  });
});
