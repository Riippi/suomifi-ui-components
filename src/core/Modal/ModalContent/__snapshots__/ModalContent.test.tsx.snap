// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic ModalContent should match snapshot 1`] = `
.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c1.fi-modal {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
}

.c1.fi-modal--no-portal {
  z-index: 100;
}

.c1.fi-modal .fi-modal_overlay {
  background-color: rgba(41,41,41,0.5);
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.fi-modal .fi-modal_content-container {
  border-radius: 4px;
  background-color: hsl(0,0%,100%);
  border-top: 4px solid hsl(212,63%,45%);
  overflow: hidden;
  max-height: calc(100vh - 50px);
  min-height: 230px;
  width: 800px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1.fi-modal--no-scroll .fi-modal_content-container {
  width: 540px;
  min-height: 190px;
}

.c1.fi-modal--small-screen .fi-modal_content-container {
  border-radius: 0;
  width: 100%;
  height: 100%;
  max-height: 100%;
}

.c3.fi-modal_content {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  max-height: 100%;
  padding: 24px 30px 50px 30px;
  overflow-y: auto;
}

.c3.fi-modal_content--no-scroll {
  overflow: hidden;
  padding: 34px 30px 10px 30px;
}

.c3.fi-modal_content--small-screen {
  padding-top: 20px;
  padding-left: 20px;
  padding-bottom: 50px;
  padding-right: 20px;
}

.c5 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  color: hsl(0,0%,16%);
}

.c5.fi-heading--h1hero {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 40px;
  line-height: 48px;
  font-weight: 600;
}

.c5.fi-heading--h1 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 40px;
  line-height: 48px;
  font-weight: 300;
}

.c5.fi-heading--h2 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 28px;
  line-height: 34px;
  font-weight: 600;
}

.c5.fi-heading--h3 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 22px;
  line-height: 28px;
  font-weight: 600;
}

.c5.fi-heading--h4 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
}

.c5.fi-heading--h5 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c5.fi-heading--h6 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c5.fi-heading--small-screen {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 300;
}

.c5.fi-heading--small-screen.fi-heading--h1hero {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 600;
}

.c5.fi-heading--small-screen.fi-heading--h1 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 300;
}

.c5.fi-heading--small-screen.fi-heading--h2 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 26px;
  line-height: 32px;
  font-weight: 600;
}

.c5.fi-heading--small-screen.fi-heading--h3 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 20px;
  line-height: 26px;
  font-weight: 600;
}

.c5.fi-heading--small-screen.fi-heading--h4 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c5.fi-heading--small-screen.fi-heading--h5 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c5.fi-heading--small-screen.fi-heading--h6 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c6.fi-modal_title {
  padding-bottom: 20px;
}

.c6.fi-modal_title--no-scroll {
  padding-bottom: 10px;
}

.c6.fi-modal_title--small-screen {
  padding-bottom: 20px;
}

<div
  aria-describedby="5_title"
  aria-modal="true"
  class="c0 c1 fi-modal fi-modal--no-portal"
  role="dialog"
>
  <div
    class="c2 fi-modal_overlay"
  >
    <div
      class="c2 fi-modal_content-container"
    >
      <div
        class="c2 fi-modal_content c3"
        data-testid="modal-content-id"
      >
        <h2
          class="c4 fi-heading c5 c6 fi-modal_title fi-heading--h3"
          id="5_title"
        >
          Test modal
        </h2>
        <p>
          Modal Content
        </p>
        <button>
          x
        </button>
      </div>
    </div>
  </div>
</div>
`;
